[KOGITO-4565] - Operator support for runtimes configuration using Pos…

vaibhavjainwiz opened the pull request:
…tgres

sutaakar commented on doc-content/kogito-docs/src/main/asciidoc/openshift/chap-kogito-deploying-on-openshift.adoc:
```diff
@@ -1168,6 +1168,88 @@ On Spring Boot, `spring.data.mongodb.uri` does not work simultaneously with host
+The {PRODUCT} Operator does not manage Postgres instances.
+If {PRODUCT} service persistence type is reactive(`kogito.persistence.type=postgresql`), then you can refer to it in the `KogitoRuntime` custom resource by configuring the following environment variables:
+
+.Required environment variables for a custom Postgresql cluster
```
```suggestion
.Required environment variables for a custom Postgresql persistence
```

vaibhavjainwiz replied:
done

sutaakar commented on doc-content/kogito-docs/src/main/asciidoc/openshift/chap-kogito-deploying-on-openshift.adoc:
```diff
@@ -1168,6 +1168,88 @@ On Spring Boot, `spring.data.mongodb.uri` does not work simultaneously with host
+
+whereas if persistence type is jdbc(`kogito.persistence.type=jdbc`), then you can refer to it in the `KogitoRuntime` custom resource by configuring the following environment variables:
+
+.Required environment variables for a custom Postgresql cluster
```
```suggestion
.Required environment variables for a custom Postgresql persistence
```

vaibhavjainwiz replied:
done

sutaakar commented on doc-content/kogito-docs/src/main/asciidoc/openshift/chap-kogito-deploying-on-openshift.adoc:
```diff
@@ -1168,6 +1168,88 @@ On Spring Boot, `spring.data.mongodb.uri` does not work simultaneously with host
+QUARKUS_DATASOURCE_JDBC_URL
+----
+
+.Example {PRODUCT} service custom resource with reactive Postgresql persistence
```
```suggestion
.Example {PRODUCT} service custom resource with JDBC Postgresql persistence
```

vaibhavjainwiz replied:
done

sutaakar commented:
We should provide also configuration for Spring Boot

radtriste reviewed (approved):
good for me with the comments from @sutaakar

vaibhavjainwiz commented:
> We should provide also configuration for Spring Boot

I had added Spring boot configuration for JDBC. For reactive, I believe Eclipse Vert.x supports both spring boot and quarkus with same envs. 
@cristianonicolai Could you please confirm same?

spolti commented on doc-content/kogito-docs/src/main/asciidoc/openshift/chap-kogito-deploying-on-openshift.adoc:
```diff
@@ -1168,6 +1168,95 @@ On Spring Boot, `spring.data.mongodb.uri` does not work simultaneously with host
+spec:
+  image: quay.io/kiegroup/process-postgresql-persistence-quarkus:persistence
+  env:
+    - name: PGUSER
```
@vaibhavjainwiz where these envs are read?

vaibhavjainwiz replied:
these envs are read by Eclipse Vert.x apis. Its a part of `kogito-addons-persistence-postgresql`. .

vaibhavjainwiz replied:
https://github.com/kiegroup/kogito-runtimes/blob/240c07f94cfb09365b2b7f42ef32d9e3f3913d73/kogito-codegen-modules/kogito-codegen-processes/src/main/java/org/kie/kogito/codegen/process/persistence/PersistenceGenerator.java#L529

briandooley commented on doc-content/kogito-docs/src/main/asciidoc/openshift/chap-kogito-deploying-on-openshift.adoc:
```diff
@@ -1168,6 +1168,95 @@ On Spring Boot, `spring.data.mongodb.uri` does not work simultaneously with host
+kogito.persistence.postgresql.connection.uri
+----
+
+whereas if persistence type is jdbc(`kogito.persistence.type=jdbc`), then you can refer to it in the `KogitoRuntime` custom resource by configuring the following environment variables:
```
```suggestion
whereas if the persistence type is jdbc(`kogito.persistence.type=jdbc`), then you can refer to it in the `KogitoRuntime` custom resource by configuring the following environment variables:
```

vaibhavjainwiz replied:
done

briandooley reviewed (changes requested):
Just one minor alteration required. 

@vaibhavjainwiz Have you made a release notes for this new change?

vaibhavjainwiz commented:
@briandooley I had updated release note

> Just one minor alteration required.
> 
> @vaibhavjainwiz Have you made a release notes for this new change?

@briandooley I had updated the release note.

hmanwani-rh reviewed (commented):
@vaibhavjainwiz I've added some tweaks to the additions. Please take a look and let me know if you're good with the changes. 

Following is the doc preview of Kogito documentation
[Kogito documentation](http://file.pnq.redhat.com/~hmanwani/BXMSDOC-KOGITO-4565/#con-kogito-operator-with-postgresql_kogito-deploying-on-openshift)
[Release notes](http://file.pnq.redhat.com/~hmanwani/BXMSDOC-KOGITO-4565/#_kogito_operator_support_for_runtimes_configuration_using_postgresql)

sutaakar commented:
PostgreSQL usage with Quarkus seems affected by https://issues.redhat.com/browse/KOGITO-5564

vaibhavjainwiz commented:
> PostgreSQL usage with Quarkus seems affected by https://issues.redhat.com/browse/KOGITO-5564

PostgreSQL usage with Spring is working fine with reactive persistence using Eclipse Vertx. I had updated documentation accordingly.

hmanwani-rh commented:
Merged PR for Kogito 1.9
